### PR TITLE
Get cookbook branch

### LINKED ISSUE #29: Create getCookbook Resolver Function
Add a getCookbook function to the Cookbook resolver.

### PR DESCRIPTION
**Pull request by RyN21**
### Type of change made:
- [x] New feature (non-breaking change which adds functionality)
- [ ] Testing - building tests
- [ ] Formatting - no new features
- [ ] Styling- no new features
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
### Description
*Add/Cleaned up cookbook resolver and cookbook types for get cookbook and cookbooks
### How should this be manually tested?
*Run `mix run priv/repo/seeds.exs` in terminal to seed data.
*Can be tested with localhost:4000/graphiql.
### What are the relevant tickets?
* Resolves #29 #18 
### Checklist
- [ ] I have tested this code
- [ ] I have updated the Readme

### CODE REVIEW

**Review by ejdelsztejn — CHANGES_REQUESTED** [review]
Hey @RyN21 - this looks great, but take a look at what is causing Semaphore's checks to fail.

File: lib/roots/cookbook.ex [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
-    has_many :recipes, Roots.Recipe
+
     belongs_to :user, Roots.User
+    has_many :recipes, Roots.Recipe
 
     timestamps()
   end
 
-  @doc false
+  def all do
```
Comment (mlynch5187) [inline lib/roots/cookbook.ex:17]: Great job following the errors in a new language to implement this!

File: mix.exs [hunk 2]
```diff
@@ -47,7 +47,10 @@ defmodule Roots.MixProject do
       {:absinthe_phoenix, "~> 2.0.0"},
       {:absinthe_plug, "~> 1.5.0"},
       {:guardian, "~> 1.0"},
-      {:dataloader, "~> 1.0.0"}
+      {:dataloader, "~> 1.0.0"},
+      {:faker, "~> 0.16", only: :test},
```
Comment (mlynch5187) [inline mix.exs:51]: Love the use of faker with Elixir.

File: test/support/factories.ex [hunk 3]
```diff
@@ -0,0 +1,18 @@
+defmodule Roots.Factory do
```
Comment (mlynch5187) [inline test/support/factories.ex:1]: Very similar to using Faker on Ruby!

File: lib/roots_web/schema/cookbook_types.ex [hunk 4]
```diff
@@ -16,19 +16,15 @@ defmodule RootsWeb.Schema.CookbookTypes do
   end
 
   object :cookbook_queries do
-    @desc "Get all cookbooks"
-    field :cookbooks, list_of(:cookbook) do
-      resolve(&Resolvers.CookbookResolver.list/3)
-    end
-
     @desc "Get specific cookbook"
     field :cookbook, :cookbook do
```
Comment (mlynch5187) [inline lib/roots_web/schema/cookbook_types.ex:20]: IIRC frontend is expecting the endpoint for getting specific cookbooks to be get_cookbook, not cookbook.

File: lib/roots_web/schema/cookbook_types.ex [hunk 5]
```diff
@@ -16,19 +16,15 @@ defmodule RootsWeb.Schema.CookbookTypes do
...
       arg(:id, non_null(:id))
       resolve(&Resolvers.CookbookResolver.show/3)
     end
-  end
 
-  # object :cookbook_mutations do
-  #
-  # end
+    @desc "Get all cookbooks"
+    field :cookbooks, list_of(:cookbook) do
```
Comment (mlynch5187) [inline lib/roots_web/schema/cookbook_types.ex:26]: I believe this should also be get_cookbooks rather than cookbooks.

**Review by mlynch5187 — CHANGES_REQUESTED** [review]
IIRC frontend is expecting the endpoints to be as such:

Get specific cookbooks -> get_cookbook
Get all cookbooks -> get_cookbooks

Other than that this looks fantastic!

**Review by mlynch5187 — APPROVED** [review]
This branch is now passing our CI, and the endpoints have been updated to match frontend's expectations. Approved!

**Review by ejdelsztejn — APPROVED** [review]

**Pull request merged**